src/generator.cs

```diff
@@ -1117,6 +1118,226 @@ static string FindProtocolInterface (Type type, MemberInfo pi)
+			break;
+			case "CGSize":
+				temp = string.Format ("NSValue.FromCGSize ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "CGVector":
+				temp = string.Format ("NSValue.FromCGVector ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "CMTimeMapping":
+				temp = string.Format ("NSValue.FromCMTimeMapping ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "CMTimeRange":
+				temp = string.Format ("NSValue.FromCMTimeRange ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "CMTime":
+				temp = string.Format ("NSValue.FromCMTime ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "MKCoordinateSpan":
+				temp = string.Format ("NSValue.FromMKCoordinateSpan ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "CLLocationCoordinate2D":
+				temp = string.Format ("NSValue.FromCLLocationCoordinate2D ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "PointF":
+				temp = string.Format ("NSValue.FromPointF ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "NSRange":
+				temp = string.Format ("NSValue.FromRange ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "RectangleF":
+				temp = string.Format ("NSValue.FromRectangleF ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "SCNMatrix4":
+				temp = string.Format ("NSValue.FromSCNMatrix4 ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "SizeF":
+				temp = string.Format ("NSValue.FromSizeF ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "UIEdgeInsets":
+				temp = string.Format ("NSValue.FromUIEdgeInsets ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "UIOffset":
+				temp = string.Format ("NSValue.FromUIOffset ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "SCNVector3":
+				temp = string.Format ("NSValue.FromVector ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
+			break;
+			case "SCNVector4":
+				temp = string.Format ("NSValue.FromVector ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
```

Comment:
that's a lot of code duplication, could not the format string be set (in the `case`) and then used outside ? that would use a single `String.Format` call (and arguments) and make it easier to change/extended (if needed).